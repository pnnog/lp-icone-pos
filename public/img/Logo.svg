<svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 278 64"><path fill="#F8FAFC" fill-rule="evenodd" d="M5.298 1.247C3.304 1.83 1.268 3.997.75 6.083c-1.713 6.919 6.638 11.456 11.526 6.262C16.872 7.462 11.77-.65 5.297 1.247Zm51.529 9.515c-3.217 1.32-3.625 6.03-.657 7.568 1.887.979 3.362.826 4.947-.51 2.292-1.934 1.852-5.39-.867-6.8-1.474-.763-2.08-.809-3.423-.258Zm-19.463 1.724c.814.101 2.146.101 2.96 0 .813-.1.147-.184-1.48-.184-1.628 0-2.294.083-1.48.184Zm-11.04.85c-14.259 2.206-22.99 6.619-22.99 11.617 0 4.309 4.461 9.215 16.375 18.005 12.062 8.9 13.65 10.456 14.019 13.732.202 1.794.038 2.326-1.294 4.2l-1.526 2.146 2.82-2.433c4.28-3.692 5.116-4.85 5.116-7.096 0-3.31-1.787-5.388-11.934-13.884-5.11-4.279-10.05-8.755-10.977-9.95-2.211-2.843-2.91-5.132-2.138-7 1.64-3.968 9.452-8 18.12-9.353 3.968-.619 3.968-.619 1.238-.619-1.502 0-4.576.285-6.83.634Zm7.993.701c-7.18.834-13.088 3.097-16.633 6.373-4.25 3.926-3.019 6.85 6.274 14.914 6 5.205 6.468 5.453 3.357 1.777-7.473-8.83-7.416-15.52.166-19.36 2.62-1.328 5.563-2.205 12.273-3.662l2.049-.444-2.049.004c-1.127.003-3.573.181-5.437.398Zm9.638 4.329c4.003.86 6.661 2.03 8.707 3.832 1.314 1.156 1.66 1.793 1.66 3.05 0 1.92-2.252 4.407-9.538 10.53-2.617 2.198-5.152 4.674-5.635 5.502-1.209 2.075-.63 3.847 1.948 5.954 1.61 1.315 1.867 1.424 1.252.527-1.827-2.664-.696-4.268 7.154-10.145 10.09-7.553 12.515-10.723 10.744-14.04-1.067-1.998-6.551-4.15-13.436-5.272-5.176-.844-6.89-.806-2.856.062Zm-2.448 1.068c4.007 1.1 6.227 2.391 7.136 4.152 1.309 2.536.75 4.74-2.19 8.648l-1.889 2.51 2-1.585c3.347-2.653 6.258-5.801 6.59-7.128.811-3.237-5.015-6.787-11.971-7.294l-3.022-.22 3.346.917Zm1.688 24.31c-1.032 1.476-.867 3.13.468 4.69.831.97 1.08 1.103.859.456-.967-2.813-1.112-3.857-.69-4.97.553-1.458.306-1.526-.637-.176Zm-12.62 9.793c.744 1.401.658 5.34-.16 7.302-.378.909-.608 1.733-.51 1.832.443.443 3.02-3.82 3.02-4.996 0-1.5-.952-3.54-2.093-4.483-.722-.597-.744-.569-.258.345Z" clip-rule="evenodd"/><path fill="#F8FAFC" d="M92.321 45.964h-6.094V19.76h6.094v26.205Zm3.86-37.581-7.313 8.583h-3.25l5.078-8.583h5.485ZM127.2 39.768c0 2.438-.474 4.08-1.422 4.927-.914.846-2.607 1.27-5.079 1.27h-15.642c-2.471 0-4.181-.424-5.129-1.27-.914-.847-1.371-2.489-1.371-4.927V25.955c0-2.438.457-4.08 1.371-4.926.948-.847 2.658-1.27 5.129-1.27h15.642c3.115 0 5.045.728 5.79 2.184.474.846.711 3.013.711 6.5l-6.094.965v-4.977h-16.455v16.607h16.455v-5.637l6.094 1.27v3.097Zm37.723 0c0 2.438-.474 4.08-1.422 4.927-.914.846-2.607 1.27-5.079 1.27h-19.298c-2.472 0-4.182-.424-5.13-1.27-.914-.847-1.371-2.489-1.371-4.927V25.955c0-2.438.457-4.08 1.371-4.926.948-.847 2.658-1.27 5.13-1.27h19.298c2.472 0 4.165.423 5.079 1.27.948.846 1.422 2.488 1.422 4.926v13.813Zm-6.095 1.27V24.431h-20.111v16.607h20.111Zm43.114 4.926h-4.164l-19.248-16.2a18.611 18.611 0 0 1-2.184-2.133c.169 1.692.254 3.03.254 4.012v14.321h-5.485V19.76h4.164l18.436 15.439c1.32 1.117 2.319 2.065 2.996 2.844-.169-2.438-.254-3.945-.254-4.52V19.759h5.485v26.205Zm33.83 0h-27.068V19.76h26.814v4.469h-20.72v5.84h12.087v4.47h-12.087v6.5h20.974v4.926ZM91.91 52.417c0 .564-.327.846-.982.846h-2.742v1.701h-1v-4.367h3.742c.655 0 .982.282.982.846v.974Zm-.982-.254v-.466c0-.147-.042-.243-.127-.288-.062-.04-.172-.059-.33-.059h-2.285v1.16h2.285c.158 0 .268-.02.33-.06.085-.045.127-.14.127-.287Zm6.773 1.955c0 .338-.068.564-.203.677-.136.113-.378.17-.728.17h-2.768c-.35 0-.593-.054-.728-.162-.135-.112-.203-.34-.203-.685V52.23c0-.344.068-.57.203-.677.135-.113.378-.17.728-.17h2.768c.35 0 .592.057.728.17.135.107.203.333.203.677v1.888Zm-.872.16v-2.234h-2.886v2.235h2.886Zm-.288-4.79-1.219 1.43h-.541l.846-1.43h.914Zm6.77 4.63c0 .338-.067.564-.203.677-.135.113-.378.17-.728.17h-2.327c-.384 0-.638-.063-.762-.187-.119-.13-.178-.384-.178-.762l.779-.144v.424h2.632v-.821h-2.319c-.35 0-.59-.054-.72-.161-.146-.119-.22-.347-.22-.686v-.398c0-.338.074-.567.22-.685.13-.108.37-.161.72-.161h2.125c.361 0 .606.056.736.17.135.112.203.343.203.693l-.753.161v-.38h-2.463v.753h2.327c.35 0 .593.056.728.169.136.107.203.333.203.677v.49Zm9.149 0c0 .338-.068.564-.204.677-.135.113-.378.17-.727.17h-2.591c-.349 0-.592-.057-.727-.17-.136-.113-.204-.339-.204-.677V52.23c0-.338.068-.564.204-.677.135-.113.378-.17.727-.17h2.591c.366 0 .612.06.736.179.13.118.195.358.195.72l-.855.143v-.38h-2.726v2.234h2.726v-.762h-1.422v-.635h2.277v1.236Zm5.928.846h-1.253l-1.498-1.354h-.821v1.354h-.863v-3.58h3.224c.559 0 .838.231.838.694v.846c0 .35-.149.565-.448.644-.102.028-.395.042-.88.042l1.701 1.354Zm-1.185-2.26v-.414c0-.192-.133-.288-.398-.288h-1.989v.99h1.989c.265 0 .398-.096.398-.288Zm6.52 2.26h-.982l-.449-.787h-2.37l-.448.787h-.872l2.124-3.58h.838l2.159 3.58Zm-1.786-1.439-.838-1.49-.804 1.49h1.642Zm7.241-.364c0 1.202-.79 1.803-2.37 1.803h-2.192v-3.58h2.412c.638 0 1.143.133 1.515.398.423.31.635.77.635 1.38Zm-.897-.042c0-.728-.415-1.092-1.245-1.092h-1.54v2.269h1.126c.536 0 .922-.063 1.159-.187.333-.18.5-.51.5-.99Zm6.452.999c0 .338-.07.564-.211.677-.136.113-.378.17-.728.17h-2.37c-.35 0-.593-.057-.728-.17-.136-.113-.203-.339-.203-.677v-2.734h.871v2.912h2.489v-2.912h.88v2.734Zm6.018.846h-.982l-.449-.787h-2.37l-.449.787h-.871l2.124-3.58h.838l2.159 3.58Zm-1.786-1.439-.838-1.49-.805 1.49h1.643Zm6.699.593c0 .338-.071.564-.212.677-.135.113-.378.17-.728.17h-2.234c-.35 0-.593-.057-.728-.17-.136-.113-.203-.339-.203-.677V52.23c0-.338.067-.564.203-.677.135-.113.378-.17.728-.17h2.234c.446 0 .723.094.83.28.073.124.11.43.11.914l-.88.127v-.66h-2.353v2.235h2.353v-.762l.88.178v.423Zm-.838 2.031c0 .525-.344.788-1.033.788-.31 0-.685-.06-1.125-.178v-.491c.457.118.809.178 1.058.178.299 0 .448-.105.448-.314 0-.163-.096-.265-.288-.304-.079-.017-.237-.023-.474-.017v-.847h.373v.39c.694 0 1.041.265 1.041.795Zm6.844-1.185h-.982l-.449-.787h-2.37l-.448.787h-.872l2.124-3.58h.838l2.159 3.58Zm-1.786-1.439-.838-1.49-.804 1.49h1.642Zm.389-3.775c-.124.621-.398.931-.821.931-.101 0-.251-.03-.448-.093a1.66 1.66 0 0 0-.449-.093c-.181 0-.299.093-.356.28h-.414c.107-.604.372-.906.795-.906.108 0 .26.03.457.093.204.062.359.093.466.093.186 0 .305-.102.355-.305h.415Zm6.835 4.368c0 .338-.068.564-.203.677-.136.113-.378.17-.728.17h-2.768c-.35 0-.592-.054-.728-.162-.135-.112-.203-.34-.203-.685V52.23c0-.344.068-.57.203-.677.136-.113.378-.17.728-.17h2.768c.35 0 .592.057.728.17.135.107.203.333.203.677v1.888Zm-.872.16v-2.234h-2.886v2.235h2.886Zm9.536.686h-3.885v-3.58h3.842v.635h-2.962v.778h1.727v.618h-1.727v.88h3.005v.67Zm6.165 0h-.813v-2.463l-1.599 2.032h-.17l-1.616-1.973v2.404h-.703v-3.58h.703l1.752 2.124 1.744-2.124h.702v3.58Zm9.407-.846c0 .338-.068.564-.204.677-.135.113-.378.17-.727.17h-2.768c-.35 0-.593-.054-.728-.162-.136-.112-.203-.34-.203-.685V52.23c0-.344.067-.57.203-.677.135-.113.378-.17.728-.17h2.768c.349 0 .592.057.727.17.136.107.204.333.204.677v1.888Zm-.872.16v-2.234h-2.887v2.235h2.887Zm6.939-1.117c0 1.202-.79 1.803-2.37 1.803h-2.192v-3.58h2.412c.638 0 1.143.133 1.515.398.424.31.635.77.635 1.38Zm-.897-.042c0-.728-.415-1.092-1.244-1.092h-1.541v2.269h1.126c.536 0 .923-.063 1.16-.187.332-.18.499-.51.499-.99Zm6.757.999c0 .338-.067.564-.203.677-.135.113-.378.17-.728.17h-2.768c-.349 0-.592-.054-.728-.162-.135-.112-.203-.34-.203-.685V52.23c0-.344.068-.57.203-.677.136-.113.379-.17.728-.17h2.768c.35 0 .593.057.728.17.136.107.203.333.203.677v1.888Zm-.872.16v-2.234h-2.886v2.235h2.886Zm6.804.686h-.601l-3.03-2.412v2.412h-.795v-3.58h.601l3.038 2.378v-2.378h.787v3.58Zm5.468-2.937h-1.736v2.937h-.871v-2.937h-1.727v-.643h4.334v.643Zm5.669 2.09c0 .34-.067.565-.203.678-.135.113-.378.17-.728.17h-2.768c-.349 0-.592-.054-.727-.162-.136-.112-.204-.34-.204-.685V52.23c0-.344.068-.57.204-.677.135-.113.378-.17.727-.17h2.768c.35 0 .593.057.728.17.136.107.203.333.203.677v1.888Zm-.871.162v-2.235h-2.887v2.235h2.887Zm6.067.685h-3.69v-3.58h.88v2.912h2.81v.668Zm5.525-.846c0 .338-.067.564-.203.677-.135.113-.378.17-.728.17h-2.768c-.35 0-.592-.054-.728-.162-.135-.112-.203-.34-.203-.685V52.23c0-.344.068-.57.203-.677.136-.113.378-.17.728-.17h2.768c.35 0 .593.057.728.17.136.107.203.333.203.677v1.888Zm-.872.16v-2.234h-2.886v2.235h2.886Zm6.745-.16c0 .338-.068.564-.203.677-.135.113-.378.17-.728.17h-2.59c-.35 0-.593-.057-.728-.17-.135-.113-.203-.339-.203-.677V52.23c0-.338.068-.564.203-.677.135-.113.378-.17.728-.17h2.59c.367 0 .612.06.736.179.13.118.195.358.195.72l-.855.143v-.38h-2.725v2.234h2.725v-.762h-1.422v-.635h2.277v1.236Zm2.365.846h-.872v-3.58h.872v3.58Zm6.171 0h-.982l-.449-.787h-2.37l-.449.787h-.871l2.124-3.58h.838l2.159 3.58Zm-1.786-1.439-.838-1.49-.805 1.49h1.643Z"/></svg>
