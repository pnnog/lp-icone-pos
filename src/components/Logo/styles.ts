import styled, { css } from 'styled-components';
import media from 'styled-media-query';
import { LogoProps } from '.';

const wrapperModifier = {
  small: () => css`
    width: 10.2rem;
    height: 3.6rem;
  `
};

export const Wrapper = styled.div<LogoProps>`
  ${({ small }) => css`
    width: 22.1rem;
    height: 4.7rem;

    ${media.greaterThan('medium')`
      width: 24.2rem;
      height: 6.2rem;
    `};

    ${!!small && wrapperModifier.small()}
  `}
`;
export const Logo = styled.img`
  width: 100%;
  height: 100%;
`;
