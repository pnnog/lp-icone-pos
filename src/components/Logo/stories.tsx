import { Meta, Story } from '@storybook/react';
import Logo, { LogoProps } from '.';

export default {
  title: 'Logo',
  component: Logo,
  argTypes: {
    small: {
      type: 'boolean'
    }
  },
  parameters: {
    backgrounds: {
      default: 'dark'
    },
    viewport: {
      defaultViewport: 'mobile1'
    }
  }
} as Meta;

export const Normal: Story<LogoProps> = (args) => <Logo {...args} />;
Normal.args = {
  small: false
};
export const Small: Story<LogoProps> = (args) => <Logo {...args} />;
Small.args = {
  small: true
};
