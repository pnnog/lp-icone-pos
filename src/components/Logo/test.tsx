import { render, screen } from '@testing-library/react';

import Logo from '.';

const props = {
  src: '/img/Logo.svg',
  alt: 'Logo branca escrito ícone. Pós graduação em odontologia'
};

describe('<Logo />', () => {
  // 1. Deve renderizar com a src correta
  it('should render Logo with correctly source', () => {
    render(<Logo />);
    expect(
      screen.getByRole('img', {
        name: /Logo branca escrito ícone. Pós graduação em odontologia/i
      })
    ).toHaveAttribute('src', props.src);
  });
});

// 2. Deve renderizar a logo pequena
it('should render normal Logo', () => {
  render(<Logo />);
  expect(screen.getByRole('img', { name: props.alt }).parentNode).toHaveStyle({
    width: '22.1rem'
  });
});

it('should render small Logo', () => {
  render(<Logo small />);
  expect(screen.getByRole('img', { name: props.alt }).parentNode).toHaveStyle({
    width: '10.2rem'
  });
});

it('should render Logo in big screen with this rules', () => {
  const { container } = render(<Logo />);
  expect(
    screen.getByRole('img', { name: props.alt }).parentNode
  ).toHaveStyleRule('width', '24.2rem', { media: '(min-width: 768px)' });

  expect(container).toMatchSnapshot();
});
