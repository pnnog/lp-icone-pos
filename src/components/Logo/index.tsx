import * as S from './styles';

export type LogoProps = {
  small?: boolean;
};
const Logo = ({ small = false }: LogoProps) => (
  <S.Wrapper small={small}>
    <S.Logo
      alt="Logo branca escrito ícone. Pós graduação em odontologia"
      src="/img/Logo.svg"
    />
  </S.Wrapper>
);

export default Logo;
