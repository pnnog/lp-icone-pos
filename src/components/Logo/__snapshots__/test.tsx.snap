// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Logo in big screen with this rules 1`] = `
.c0 {
  width: 22.1rem;
  height: 4.7rem;
}

.c1 {
  width: 100%;
  height: 100%;
}

@media (min-width:768px) {
  .c0 {
    width: 24.2rem;
    height: 6.2rem;
  }
}

<div>
  <div
    class="c0"
  >
    <img
      alt="Logo branca escrito ícone. Pós graduação em odontologia"
      class="c1"
      src="/img/Logo.svg"
    />
  </div>
</div>
`;
